ImagePicker/UICollectionView+CellRegistrator.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  UICollectionView+CellRegistrator.swift
+//  ImagePicker
+//
+//  Created by Anna Shirokova on 14/06/2018.
+//  Copyright © 2018 Inloop. All rights reserved.
+//
+
+import Foundation
+
+
+extension UICollectionView {
+    ///
+    /// Used by datasource when registering all cells to the collection view. If user
+    /// did not register custom cells, this method registers default cells
+    ///
+    func apply(registrator: CellRegistrator, cameraMode: CaptureSettings.CameraMode) {
+        registerActionItems(registrator: registrator)
+        registerCameraItem(registrator: registrator, cameraMode: cameraMode)
+        registerAssetItems(registrator: registrator)
+    }
+}
+
+// MARK: - Helper private methods
+private extension UICollectionView {
+    func registerActionItems(registrator: CellRegistrator) {
+        if registrator.hasUserRegisteredActionCell {
+            register(nibsData: registrator.actionItemNibsData?.map { $1 })
```

Comment:
both mappings done using $1 ? Seems weird

Reply:
That's not my code, but as I can see, it's correct: `actionItemNibsData` and `actionItemClassesData` are dictionaries:
```swift
    var actionItemNibsData: [Int: (UINib, String)]?
    var actionItemClassesData: [Int: (UICollectionViewCell.Type, String)]?
```
and `register` methods are getting corresponding values, that's why there is $1